botogram/objects/chats.py

```diff
@@ -257,9 +269,105 @@ def unban(self, user):
 
         self._api.call("unbanChatMember", {
             "chat_id": self.id,
-            "user_id": user,
+            "user_id": user
         })
 
+    def kick(self, user, time=None):
+        if self.type == "private":
+            raise RuntimeError("This chat is a private!")
+        if isinstance(user, User):
+            user = user.id
+
+        if time is None:
+            self._api.call("unbanChatMember", {
+                "chat_id": self.id,
+                "user_id": user
+            })
+        else:
+            if isinstance(time, dt):
+                time = mktime(time.timetuple())
+            self._api.call("kickChatMember", {
+                "chat_id": self.id,
+                "user_id": user,
+                "until_date": time
+            })
+
+    def permissions(self, user):
+        return Permissions(user, self)
+
+
+class Permissions:
+    def __init__(self, user, chat):
+        if chat.type not in ("group", "supergroup"):
+            raise RuntimeError("This chat is not a group or a supergroup!")
+        # Accept also an instance of `User`
+        self.chatid = chat.id
+        self._api = chat._api
+        if isinstance(user, User):
+            self.user = user.id
+        else:
+            self.user = user
+        infouser = self._api.call("getChatMember", {
+            "chat_id": self.chatid,
+            "user_id": self.user},
+                                  expect=ChatMember)
+
+        try:
+            self._until_date = infouser.until_date
+        except AttributeError:
+            self._until_date = 0
+        self.until_date = self._until_date
+
+        try:
+            self._send_messages = infouser.can_send_messages
+        except AttributeError:
+            self._send_messages = True
+        self.send_messages = self._send_messages
+
+        try:
+            self._send_media_messages = infouser.can_media_messages
+        except AttributeError:
+            self._send_media_messages = True
+        self.send_media_messages = self._send_media_messages
+
+        try:
+            self._send_other_messages = infouser.can_send_other_messages
+        except AttributeError:
+            self._send_other_messages = True
+        self.send_other_messages = self._send_other_messages
+
+        try:
+            self._add_web_page_previews = infouser.can_add_web_page_previews
+        except AttributeError:
+            self._add_web_page_previews = True
+        self.add_web_page_previews = self._add_web_page_previews
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.save()
```

Comment:
I don't remember the exact behavior of context managers, are you sure that if an exception is raised inside the context manager it's propagated correctly? Also, we should save only if there was no exception.